### PR TITLE
Test: Improve coverage

### PR DESCRIPTION
**Pull request by fisker**
Improve test coverage https://codecov.io/gh/prettier/prettier/pull/8811

- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-graphql/printer-graphql.js [hunk 1]
```diff
@@ -275,10 +275,6 @@ function genericPrint(path, options, print) {
       ]);
     }
 
-    case "TypeExtensionDefinition": {
```
Comment (fisker) [inline src/language-graphql/printer-graphql.js:278]: This AST is no longer used.

https://github.com/graphql/graphql-js/blob/58aaa1df30b501b67d6986ea2fc1a8b95ac9fbca/src/language/ast.js#L144

File: src/utils/line-column-to-index.js [hunk 2]
```diff
@@ -5,9 +5,6 @@ module.exports = function (lineColumn, text) {
   let index = 0;
   for (let i = 0; i < lineColumn.line - 1; ++i) {
     index = text.indexOf("\n", index) + 1;
-    if (index === -1) {
-      return -1;
-    }
```
Comment (fisker) [inline src/utils/line-column-to-index.js:10]: Improssible to be `-1`

File: src/language-css/loc.js [hunk 3]
```diff
@@ -4,13 +4,11 @@ const lineColumnToIndex = require("../utils/line-column-to-index");
 const { getLast, skipEverythingButNewLine } = require("../common/util");
 
 function calculateLocStart(node, text) {
-  if (node.source) {
-    if (typeof node.source.sourceIndex === "number") {
-      return node.source.sourceIndex; // value-* nodes have this
```
[the remark this answers is not in the document]
Comment (thorn0) [reply src/language-css/loc.js:9]: Right. It should be `node.sourceIndex`, not `node.source.sourceIndex`.

[the remark this answers is not in the document]
Comment (fisker) [reply src/language-css/loc.js:9]:
```console
> require('postcss-values-parser')('1px').sourceIndex
undefined
```

This is undefined too

File: src/language-html/printer-html.js [hunk 4]
```diff
@@ -448,10 +438,8 @@ function genericPrint(path, options, print) {
         ]),
       ]);
     }
-    case "yaml":
-    case "toml":
-      return concat(replaceEndOfLineWith(node.raw, literalline));
```
Comment (fisker) [inline src/language-html/printer-html.js:453]: Changed to `front-matter`,  handled/tested.

File: src/language-css/loc.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (thorn0) [reply src/language-css/loc.js:9]:
```console
> require('postcss-values-parser')('1px').parse().nodes[0].nodes[0].sourceIndex
0
```

File: src/language-js/clean.js [hunk 5]
```diff
@@ -44,30 +44,6 @@ function clean(ast, newObj, parent) {
...
-  // (TypeScript) ignore empty `specifiers` array
-  if (
-    ast.type === "TSNamespaceExportDeclaration" &&
-    ast.specifiers &&
-    ast.specifiers.length === 0
-  ) {
-    delete newObj.specifiers;
-  }
-
   // We convert <div></div> to <div />
```
Comment (thorn0) [inline src/language-js/clean.js:47]: We don't.

**Comment by fisker** [comment]
@thorn0 I've sending coverage from windows, any idea how to test these missing  lines in this file?

https://codecov.io/gh/prettier/prettier/pull/8811/src/src/main/core.js?before=src/main/core.js

**Comment by fisker** [comment]
I think I've tested all reachable lines. 

Result: https://codecov.io/gh/prettier/prettier/pull/8811/tree/src

Every dir in `src` is 100% covered.

Except:

1. `src/language-css` 

Some note type missing, but I'm not sure is it safe to remove, maybe investigate later
Uncovered line for front-matter is refactoring in https://github.com/prettier/prettier/pull/8595


2. `src/language-js`

This part is a little complicated, I won't fix it in this PR.

3. `src/main`

All covered, but `\r` test only run on windows, and not sending coverage.

File: src/language-js/utils.js [hunk 6]
```diff
@@ -688,7 +688,7 @@ function isLastStatement(path) {
   const body = (parent.body || parent.consequent).filter(
     (stmt) => stmt.type !== "EmptyStatement"
   );
-  return body && body[body.length - 1] === node;
+  return body[body.length - 1] === node;
```
Comment (fisker) [inline src/language-js/utils.js:691]: `body` is already `Array` in line above.

**Comment by fisker** [comment]
Most of they are unreachable. At least hard to test.

**Comment by fisker** [comment]
I already did the best I can do, consider they are TODOs.

**Pull request merged**